### DISCUSSION TITLE
Scrolling in mouse-based programs should probably only send a single scroll signal

### DISCUSSION THREAD

**Discussion by liquidev** [ghostty-org/ghostty#3955] [Feature Requests, Ideas]
It appears that when running under ghostty, [Helix](https://helix-editor.com) receives multiple scroll signals when you use the scroll wheel. This causes the screen to visibly update several times before it settles into place (sort of "animating" the scroll in an ugly way,) and also makes the default scroll height a little too big (because Helix scrolls 3 lines by default, couple that with ghostty's default which is 2 I believe, so you get 6 lines—which is unlike how any other terminal I've tried behaves.)

I unfortunately don't have a video because the effect happens too fast to be easy to capture... but it's pretty easy to reproduce locally in Helix.

**Comment by mitchellh** [comment]
This isn't a bug, this is to handle inertial scrolling with macOS. Feel free to Git blame that code and see more reasoning. I'm willing to look into how other terminals handle this better though.

**Reply by 00-kat** [reply]
I wasn't able to find the reasoning either in the source code or through `git blame`ing; but I wanted to ask, is `mouse-scroll-multiplier` affecting the number of scroll events sent also related to inertial scrolling with macOS, or is it only the default value of two events that matters?  I'm not able to figure out if that is a separate issue or not.

**Comment by prurigro** [comment]
So in linux I'm getting 4x the scroll- setting the mouse-scroll-multiplier to 0.25 makes it behave like other terminals. Would it be worth not applying this logic when not in macos?

**Reply by rhodes-b** [reply]
The main branch has changed this #6052

**Reply by ilyagr** [reply]
At least on Mac, this doesn't seem to be fixed. https://github.com/ghostty-org/ghostty/discussions/7577.

**Comment by janos-r** [comment]
I experience a similar multiplication in Helix on linux. I had to set the mouse-scroll-multiplier to 0.5 to jump only 3 line per scroll in helix as I was used to. In ghostty it scrolls one line per scroll this way. Which is not ideal but acceptable for now. I wish there was a better solution for this. I wonder if the problem is on Helixes end that they multiply scrolls?

**Reply by ChrHorn** [reply]
> I experience a similar multiplication in Helix on linux. I had to set the mouse-scroll-multiplier to 0.5 to jump only 3 line per scroll in helix as I was used to. In ghostty it scrolls one line per scroll this way. Which is not ideal but acceptable for now. I wish there was a better solution for this. I wonder if the problem is on Helixes end that they multiply scrolls?

Neovim has the same issue. Both Helix and Neovim scroll 18 lines per tick with default settings.

There is https://github.com/ghostty-org/ghostty/pull/8907, but it seems to have stalled.

**Comment by supermarin** [comment]
+1 resurrecting this one - on sway and hyprland observing 4x scroll within tmux or nvim
